src/DurableTask.Netherite/StorageLayer/Faster/StoreWorker.cs

```diff
@@ -281,6 +282,42 @@ internal enum CheckpointTrigger
             Idle
         }
 
+        void LogCheckpointStats()
+        {
+            long inputQueuePositionLag = this.getInputQueuePositionLag();
+            this.CheckpointDue(out CheckpointTrigger trigger, out long? compactUntil);
+            var reportNullableTaskStatus = (Task? t) =>
+            {
+                if (t == null)
+                {
+                    return "is null";
+                }
+                else
+                {
+                    return t.IsCompleted ? "is completed" : "is not completed";
+                }
+            };
+
+            var checkpointDueLog = $"Checkpoint statistics: current checkpoint trigger={trigger}, " +
+                $"LastCheckpointedCommitLogPosition={this.lastCheckpointedCommitLogPosition}, " +
+                $"MaxNumberBytesBetweenCheckpoints={this.partition.Settings.MaxNumberBytesBetweenCheckpoints}, " +
+                $"CommitLogPosition={this.CommitLogPosition}, " +
+                $"NumberEventsSinceLastCheckpoint={this.numberEventsSinceLastCheckpoint}, " +
+                $"MaxNumberEventsBetweenCheckpoints={this.partition.Settings.MaxNumberEventsBetweenCheckpoints}, " +
+                $"InputQueuePositionLag={inputQueuePositionLag}," +
+                $"TimeOfNextIdleCheckpoint={this.timeOfNextIdleCheckpoint}, " +
+                $"TimeOfFirstRefusedCheckpoint={this.timeOfFirstRefusedCheckpoint}, " +
+                $"PendingCompaction status={reportNullableTaskStatus(this.pendingCompaction)}, " +
+                $"PendingIndexCheckpoint status={reportNullableTaskStatus(this.pendingIndexCheckpoint)}, " +
+                $"PendingStoreCheckpoint status={reportNullableTaskStatus(this.pendingStoreCheckpoint)}";
+            this.traceHelper.FasterProgress(checkpointDueLog);
```

Comment:
Are we expected to write this trace always or only when certain trace levels are configured? If the latter, and if we may end up writing this frequently, consider making this into a lambda function that gets passed to `this.traceHelper.FasterProgress` so that we execute this string generation only when needed (and put less pressure on the .NET GC).

Reply:
> Are we expected to write this trace always or only when certain trace levels are configured?

As per the definition of `FasterProgress`, this is only logged when the trace level is at least `Debug`. This is the case by default.

> If the latter, and if we may end up writing this frequently, consider making this into a lambda function that gets passed to this.traceHelper.FasterProgress

Good idea, I've incorporated that feedback in my next commit.